### DISCUSSION TITLE
Get iPhone Marketing Name

### DISCUSSION THREAD

**Discussion by swernimo** [ionic-team/capacitor#4237] [Q&A]
I'm trying to add analytics to my ionic 5, Angular 10, Capacitor 2 app. I'm using the Device Plugin to get the device information. When I call Device.Model I'm only getting "iPhone". Is it possible to get the marketing name (ie iPhone 6s, Xr, 12 mini, etc) for iphones? I want to know which iPhone is most popular among my users, not just that they're on an iPhone, I can already tell that by platform.

**Comment by jcesarmobile** [comment]
There is no native API for getting the marketing name, only the official model (i.e. iPhone13,1 for iPhone 12 mini)

You can create a plugin that uses [DeviceKit](https://github.com/devicekit/DeviceKit) for getting the marketing name, but what it does is just to convert the official model to the marketing name by hardcoding values.

**Comment by finnmerlett** [comment]
I wrote a solution here: https://github.com/ionic-team/capacitor/discussions/4353#discussioncomment-1461789
